File: 12nd/pro_4.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        int i=0,j=0;
        for(i=3; i <= (int) (brown + 4) / 2; i++){
            j = ((4+brown)/2)-i;
            if(i>=j && i*j==yellow+brown) break;
        }
        answer[0]=i;
        answer[1]=j;
        return answer;
    }
}

File: bfs, dfs/pro_4.cpp
#include <string>
#include <vector>

using namespace std;

bool visited[200]={0,};

void dfs(int n, vector<vector<int>> computers){
    visited[n]=1;
    for(int i=0; i<computers[n].size(); i++){
        if(!visited[i]&&computers[n][i]) dfs(i,computers);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i=0; i<computers.size(); i++){
        if(visited[i]) continue;
        dfs(i,computers);
        answer++;
    }
    return answer;
}